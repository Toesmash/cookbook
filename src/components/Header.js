import React from 'react';
import { Link } from 'react-router-dom';

const Header = () => (
  <header>
    <Link to="/"><h1>CookBook</h1></Link>
    <Link to="/create">Create Recipe</Link>
  </header>
);

export default Header;