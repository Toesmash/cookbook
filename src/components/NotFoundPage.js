import React from 'react';
import { Link } from 'react-router-dom';

const NotFound = () => (
  <div className='notfound__msg'>
    <h1>Page not found :(</h1>
    <p><Link to="/">Go home</Link></p>
  </div>
);

export default NotFound;