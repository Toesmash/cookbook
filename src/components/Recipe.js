import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { removeRecipe, likeRecipe, dislikeRecipe } from '../redux/actions/recipes';


export class Recipe extends React.Component {
  onRemove = () => {
    this.props.removeRecipe(this.props.recipe.id);
    this.props.history.push('/');
  };
  onLike = () => {
    this.props.likeRecipe(this.props.recipe.id)
  };
  onDislike = () => {
    this.props.dislikeRecipe(this.props.recipe.id)
  };

  render() {
    const { title, ingredients, instructions, id } = this.props.recipe;
    return (
      <div>
        <p>{title}</p>
        <p>{ingredients}</p>
        <p>{instructions}</p>
        <button><Link to={`/edit/${id}`}>Edit Recipe</Link></button>
        <button onClick={this.onRemove} >Delete Recipe</button>
        <button onClick={this.onLike}>Like</button>
        <button onClick={this.onDislike}>Dislike</button>
      </div>
    );
  };
};

const mapDispatchToProps = (dispatch) => ({
  removeRecipe: (id) => dispatch(removeRecipe(id)),
  likeRecipe: (id) => dispatch(likeRecipe(id)),
  dislikeRecipe: (id) => dispatch(dislikeRecipe(id))
});

const mapStateToProps = (store, props) => ({
  recipe: store.recipes.find((item) => {
    return item.id == props.match.params.id;
  })
});

export default connect(mapStateToProps, mapDispatchToProps)(Recipe);