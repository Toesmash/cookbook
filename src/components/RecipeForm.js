import React from 'react';
import * as Yup from 'yup';
import { withFormik, Form, Field } from 'formik';

export const RecipeForm = ({ errors, touched }) => {
  return (
    <div>
      <Form>
        <label>Title</label>
        <Field type="text" name="title" component="input" />
        {errors.title && touched.title && <div>{errors.title}</div>}

        <label>Portions</label>
        <Field type="number" name="feeds" component="input" />
        {errors.feeds && touched.feeds && <div>{errors.feeds}</div>}

        <label>Preparation time in minutes</label>
        <Field type="number" name="preparation" component="input" />
        {errors.preparation && touched.preparation && <div>{errors.preparation}</div>}

        <label>Ingredients</label>
        <Field type="text" name="ingredients" component="textarea" />
        {errors.ingredients && touched.ingredients && <div>{errors.ingredients}</div>}

        <label>Instructions</label>
        <Field type="text" name="instructions" component="textarea" />
        {errors.instructions && touched.instructions && <div>{errors.instructions}</div>}

        <button type='submit'>Submit</button>
      </Form>
    </div>
  );
}

const schema = {
  title: Yup.string().required('Required field').matches(/^[a-zA-Z0-9,\.\-\/:\(\)]*$/, 'No special characters allowed'),
  feeds: Yup.number().required('Required field').min(1, 'Minimum number of portions is 1').max(8, 'Maximum number of portions is 8'),
  preparation: Yup.number().required('Required field').min(10, 'Minimum preparation time is 10minutes').max(300, 'Maximum preparation time is 5 hours'),
  ingredients: Yup.string().required('Required field').matches(/^[a-zA-Z0-9,\.\-\/:\(\)\n\s]*$/, 'No special characters allowed'),
  instructions: Yup.string().required('Required field').matches(/^[a-zA-Z0-9,\.\-\/:\(\)\n\s]*$/, 'No special characters allowed')
};

export const FormikRecipeForm = withFormik({
  mapPropsToValues({ recipe }) {
    return {
      title: recipe ? recipe.title : '',
      feeds: recipe ? recipe.feeds : '',
      preparation: recipe ? recipe.preparation : '',
      ingredients: recipe ? recipe.ingredients.join('\n') : '',
      instructions: recipe ? recipe.instructions.join('\n') : ''
    }
  },
  validationSchema: Yup.object().shape(schema),
  handleSubmit(values, { props }) {
    props.onSubmit({
      ...values,
      ingredients: values.ingredients.split('\n').filter(item => item),
      instructions: values.instructions.split('\n').filter(item => item)
    });
  }

})(RecipeForm);

export default FormikRecipeForm;




//   onSubmit = (e) => {
//     e.preventDefault();
//     if (!this.state.title || !this.state.ingredients || !this.state.instructions) {
//       this.setState(() => ({ error: 'Please set correct' }))
//     }
//     else {
//       this.setState(() => ({
//         error: ''
//       }))
//       this.props.onSubmit({
//         title: this.state.title,
//         feeds: this.state.feeds,
//         preparation: this.state.preparation,
//         ingredients: this.state.ingredients.filter((item) => (item)),
//         instructions: this.state.instructions.filter((item) => (item))
//       })
//     }
//   }





// export default class RecipeForm extends React.Component {
//   constructor(props) {
//     super(props);
//     this.state = {
//       title: props.recipe ? props.recipe.title : '',
//       feeds: props.recipe ? props.recipe.feeds : '',
//       preparation: props.recipe ? props.recipe.preparation : '',
//       ingredients: props.recipe ? props.recipe.ingredients : [],
//       instructions: props.recipe ? props.recipe.instructions : [],
//       error: ''
//     };
//   }

//   onTitleChange = (e) => {
//     const title = e.target.value;
//     this.setState(() => ({ title }))
//   }
//   onFeedsChange = (e) => {
//     const feeds = e.target.value;
//     this.setState(() => ({ feeds }))

//   }
//   onPrepChange = (e) => {
//     const preparation = e.target.value;
//     this.setState(() => ({ preparation }))

//   }
//   onInstructionChange = (e) => {
//     const instructions = e.target.value;
//     this.setState(() => ({ instructions: instructions.split("\n") }))
//   }
//   onIngredientsChange = (e) => {
//     const ingredients = e.target.value;
//     this.setState(() => ({ ingredients: ingredients.split("\n") }))
//   }

//   onSubmit = (e) => {
//     e.preventDefault();
//     if (!this.state.title || !this.state.ingredients || !this.state.instructions) {
//       this.setState(() => ({ error: 'Please set correct' }))
//     }
//     else {
//       this.setState(() => ({
//         error: ''
//       }))
//       this.props.onSubmit({
//         title: this.state.title,
//         feeds: this.state.feeds,
//         preparation: this.state.preparation,
//         ingredients: this.state.ingredients.filter((item) => (item)),
//         instructions: this.state.instructions.filter((item) => (item))
//       })
//     }
//   }

//   render() {
//     return (
//       <div>
//         {this.state.error && <p>{this.state.error}</p>}
//         <form onSubmit={this.onSubmit}>
//           <input
//             autoFocus
//             type="text"
//             name="title"
//             placeholder="Title"
//             value={this.state.title}
//             onChange={this.onTitleChange}
//           />
//           <input
//             type="text"
//             name="feeds"
//             placeholder="2"
//             value={this.state.feeds}
//             onChange={this.onFeedsChange}
//           />
//           <input
//             type="text"
//             name="preparation"
//             placeholder="60"
//             value={this.state.preparation}
//             onChange={this.onPrepChange}
//           />
//           <textarea
//             placeholder="Ingredients"
//             value={this.state.ingredients.join("\n")}
//             onChange={this.onIngredientsChange}
//           />
//           <textarea
//             placeholder="Instructions"
//             value={this.state.instructions.join("\n")}
//             onChange={this.onInstructionChange}
//           />
//           <button>Submit</button>
//         </form>
//       </div>
//     );
//   }
// }


