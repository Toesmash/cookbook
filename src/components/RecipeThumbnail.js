import React from 'react';
import { Link } from 'react-router-dom';

const RecipeThumbnail = ({ id, title, rating }) => {
  return (
    <div>
      <p><Link to={`/view/${id}`}>{title}</Link></p>
      {rating.like && <span>like</span>}
      {rating.dislike && <span>dislike</span>}
    </div>
  );
}

export default RecipeThumbnail;