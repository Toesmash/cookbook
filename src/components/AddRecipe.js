import React from 'react';
import { connect } from 'react-redux';
import RecipeForm from './RecipeForm';
import { addRecipe } from '../redux/actions/recipes';

export class AddRecipe extends React.Component {
  onSubmit = (recipe) => {
    this.props.addRecipe(recipe);
    this.props.history.push('/');
  };

  render() {
    return (
      <div>
        <RecipeForm
          onSubmit={this.onSubmit}
        />
      </div>
    );
  };
}

const mapDispatchToProps = (dispatch) => ({
  addRecipe: (recipe) => dispatch(addRecipe(recipe))
});

export default connect(undefined, mapDispatchToProps)(AddRecipe);