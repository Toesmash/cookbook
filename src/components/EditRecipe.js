import React from 'react';
import { connect } from 'react-redux';
import RecipeForm from './RecipeForm';
import { editRecipe } from '../redux/actions/recipes';

const Edit = (props) => {
  return (
    <div>
      <RecipeForm
        recipe={props.recipe}
        onSubmit={(recipe) => {
          props.dispatch(editRecipe(props.recipe.id, recipe));
          props.history.push('/')
        }}
      />
    </div>
  );
}

const mapStateToProps = (store, props) => ({
  recipe: store.recipes.find((item) => {
    return item.id == props.match.params.id;
  })
});

export default connect(mapStateToProps)(Edit);