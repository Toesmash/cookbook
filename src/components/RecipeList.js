import React from 'react';
import { connect } from 'react-redux';
import RecipeThumbnail from './RecipeThumbnail';


export const RecipeList = (props) => {
  return (
    <div>
      <h1>Title</h1>
      {
        props.recipes.length === 0 ? <p>No recipes</p> : (
          props.recipes.map((recipe) => (
            <RecipeThumbnail key={recipe.id} {...recipe} />))
        )
      }
    </div>
  );
}

const mapStateToProps = (store) => ({
  recipes: store.recipes
});

export default connect(mapStateToProps)(RecipeList);