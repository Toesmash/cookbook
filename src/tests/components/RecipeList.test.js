import { shallow } from 'enzyme';
import React from 'react';
import { RecipeList } from '../../components/RecipeList';
import dummyData from '../../redux/fixtures/dummyData';

test('Render RecipeList with data', () => {
  const wrapper = shallow(<RecipeList recipes={dummyData} />);
  expect(wrapper).toMatchSnapshot();
});

test('Render RecipeList without data', () => {
  const wrapper = shallow(<RecipeList recipes={[]} />);
  expect(wrapper).toMatchSnapshot();
});