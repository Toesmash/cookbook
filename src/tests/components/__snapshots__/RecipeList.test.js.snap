// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render RecipeList with data 1`] = `
<div>
  <h1>
    Title
  </h1>
  <RecipeThumbnail
    feeds={4}
    id="1"
    ingredients={
      Array [
        "1kg zemiaky",
        "300g hladka muka",
        "sol",
        "250g bryndza",
        "100g slaninka",
      ]
    }
    instructions={
      Array [
        "Ocistit zemiaky, nastruhat, pridat muku a sol",
        "Zamiesit cesto",
        "Vlozit do vriacej vody",
        "Zmiesat s bryndzou",
      ]
    }
    key="1"
    preparation={60}
    rating={
      Object {
        "dislike": false,
        "like": false,
      }
    }
    title="halusky"
  />
  <RecipeThumbnail
    feeds={4}
    id="2"
    ingredients={
      Array [
        "Parky",
        "Kecup",
      ]
    }
    instructions={
      Array [
        "Vlozit parky do vriacej vody",
        "Pridat kecup",
      ]
    }
    key="2"
    preparation={20}
    rating={
      Object {
        "dislike": false,
        "like": false,
      }
    }
    title="Parky"
  />
</div>
`;

exports[`Render RecipeList without data 1`] = `
<div>
  <h1>
    Title
  </h1>
  <p>
    No recipes
  </p>
</div>
`;
